src/commands/distribute/release.ts

```diff
@@ -144,83 +120,34 @@ export default class ReleaseBinaryCommand extends AppCommand {
+        app.ownerName, app.appName, cb));
     } catch (error) {
       throw failure(ErrorCodes.Exception, `failed to create release upload for ${this.filePath}`);
     }
 
     return createReleaseUploadRequestResponse.result;
   }
 
-  private uploadFileToUri(uploadUrl: string, fileBuffer: Buffer, filename: string): Promise<void> {
-    debug("Uploading the release binary");
-    return new Promise<void>((resolve, reject) => {
-      Request.post({
-        formData: {
-          ipa: {
-            options: {
-              filename,
-              contentType: "application/octet-stream"
-            },
-            value: fileBuffer
-          }
-        },
-        url: uploadUrl
-      })
-      .on("error", (error) => {
-        reject(failure(ErrorCodes.Exception, `release binary uploading failed: ${error.message}`));
-      })
-      .on("response", (response) => {
-        if (response.statusCode < 400) {
-          resolve();
-        } else {
-          reject(failure(ErrorCodes.Exception, `release binary file uploading failed: HTTP ${response.statusCode} ${response.statusMessage}`));
-        }
-      });
-    });
-  }
-
-  private async finishReleaseUpload(client: AppCenterClient, app: DefaultApp, uploadId: string): Promise<string> {
-    let finishReleaseUploadRequestResponse: ClientResponse<models.ReleaseUploadEndResponse>;
-    try {
-      finishReleaseUploadRequestResponse = await out.progress("Finishing release upload...",
-        clientRequest<models.ReleaseUploadEndResponse>((cb) => client.releaseUploads.complete(uploadId, app.ownerName, app.appName, "committed", cb)));
-    } catch (error) {
-      throw failure(ErrorCodes.Exception, `failed to finish release upload for ${this.filePath}`);
-    }
-
-    return finishReleaseUploadRequestResponse.result.releaseUrl;
-  }
-
-  private async abortReleaseUpload(client: AppCenterClient, app: DefaultApp, uploadId: string): Promise<void> {
-    let abortReleaseUploadRequestResponse: ClientResponse<models.ReleaseUploadEndResponse>;
+  private async processUploadedFile(client: AppCenterClient, app: DefaultApp, assetId: string): Promise<models.UploadProgress> {
+    let uploadProgressResponse: ClientResponse<models.UploadProgress>;
     try {
-      abortReleaseUploadRequestResponse = await out.progress("Aborting release upload...",
-        clientRequest<models.ReleaseUploadEndResponse>((cb) => client.releaseUploads.complete(uploadId, app.ownerName, app.appName, "aborted", cb)));
+      uploadProgressResponse = await clientRequest<models.UploadProgress>((cb) => client.releaseUploads.get(assetId, 
+        app.ownerName, 
+        app.appName, 
+        cb));
     } catch (error) {
-      throw new Error(`HTTP ${abortReleaseUploadRequestResponse.response.statusCode} - ${abortReleaseUploadRequestResponse.response.statusMessage}`);
+      throw failure(ErrorCodes.Exception, `Processing the uploaded app file failed: ${error}.`);
```

Comment:
I would change the error to "Failed to process the package." and again separate 500s from 400s. On 500s the the expectation from the user is ping us in intercom, on 400s we probably want them to change something in the package.